### ISSUE TITLE
calling fire_and_forget and/or client.submit multiple times but it's not running in parallel 

### ISSUE THREAD

**Issue by deKeijzer** [dask/dask#4032]
Hi there,

I'm new to dask and have been spitting through the documentation this weekend, to be able to parallelize the reading, cleaning and saving of multiple dataframes. I managed to get most of it to work, apart from saving Pandas dataframes in parallel. The `fire_and_forget` and `client.submit` functions seem to execute this part of the code on only one thread of my CPU, while Dask uses 8 threads for everything else. 
Maybe this is an issue from my side, but i have not been able to find anything about this in the Futures and Delayed documentation (read through both and almost tried all examples which are given). 

The complete notebook that i am using can be found [here](https://github.com/deKeijzer/KB-74-OPSCHALER/blob/master/Personal_folders/Brian/Dask/Dask%20test%20setup.ipynb). Basically, the loop below is creating all the dataframes. All functions used in here are delayed functions, so it runs quick on lots of large dataframes.

```
%%time

client.restart()

weather = read_weather_data()

file_paths, dwelling_ids = smartmeter_data()

file_paths = file_paths[:10]

dfs_hour = []
dfs_10s = []
dfs_nan_table_10s = []
dfs_nan_table_hour = []

for i, path in enumerate(file_paths):
    dwelling_id = dwelling_ids[i]
    
    smart, gas = clean_prepare_smart_gas(path, dwelling_id)
    
    # client.persist: Start computing these variables and keep them in memory
    smart = client.persist(smart)
    gas = client.persist(gas)

    smart, gas, weather = resample_dfs(smart, gas, weather)
    
    df_hour = create_hour_df(smart, gas, weather, dwelling_id)
    df_10s = create_10s_df(smart, gas, weather, dwelling_id)
    
    df_hour = client.persist(df_hour)
    df_10s = client.persist(df_10s)
    
    #Slow, plus low cpu usage...
    #fig = plot_nans(df_10s, dwelling_id+' 10s sample rate', '1T')
    df_nan_table_10s = df_nan_checker(df_10s, 0)
    df_nan_table_hour = df_nan_checker(df_hour, 0)
    
    dfs_hour.append(df_hour)
    dfs_10s.append(df_10s)
    dfs_nan_table_10s.append(df_nan_table_10s)
    dfs_nan_table_hour.append(df_nan_table_hour)
```

Since i have not been able to save each `df_10s, df_hour, df_nan_table_10s, df_nan_table_hour` from within the loop, i tried doing it from outside the loop:

```
for i in range (len(dfs_10s)):
    df = dfs_10s[i]
    future = client.submit(save_df_unprocessed, df, (dwelling_ids[i]+'_10s'))
    fire_and_forget(future)
```
Or with
```
for i in range (len(dfs_10s)):
    df = dfs_10s[i].compute()
    future = client.submit(save_df_unprocessed, df, (dwelling_ids[i]+'_10s'))
```
They both save the dataframes, but not in parallel. Were `save_df_unprocessed` is not a delayed function.

What's the proper way to do this?

**Comment by mrocklin** [comment]
Possible duplicate of
https://stackoverflow.com/questions/52572542/dask-in-dask-distributed-how-to-force-run-delayed-functions-some-more-dask
.  Please delete one of them.

Also, I suspect that you will get a more speedy answer if you provide a
minimal reproducible example.

See http://docs.dask.org/en/latest/support.html#asking-for-help

**Comment by deKeijzer** [comment]
@mrocklin Just deleted the stack overflow one, since this one is more detailed i guess. Thanks for the fast reply.

**Comment by mrocklin** [comment]
Great.  Let us know if you get a minimal reproducible example.

**Comment by mrocklin** [comment]
Closing due to lack of response.
